src/util/timer.cpp

```diff
@@ -80,30 +80,31 @@ mixxx::Duration SuspendableTimer::elapsed(bool report) {
 
 GuiTickTimer::GuiTickTimer(QObject* pParent)
         : QObject(pParent),
-          m_pGuiTick50ms(new ControlProxy(
-              "[Master]", "guiTick50ms", this)),
+          m_pGuiTick20ms(std::make_unique<ControlProxy>(
+              "[Master]", "guiTick20ms", this)),
           m_bActive(false) {
-    m_pGuiTick50ms->connectValueChanged(SLOT(slotGuiTick50ms(double)));
 }
 
 GuiTickTimer::~GuiTickTimer() {
 }
 
 void GuiTickTimer::start(mixxx::Duration duration) {
+    m_pGuiTick20ms->connectValueChanged(SLOT(slotGuiTick20ms(double)));
```

Comment:
Keep in mind this is not called all the time. It is only called at the start of mouse input on a knob or slider.

Reply:
It locks the sender and receiver's signal/slot list, not a global mutex, unless you're remembering something I'm not? Since this isn't connected to anything in the audio thread I don't think it could cause audio skips.

Reply:
Oh yea, you're right -- all QObject's share a pool of mutexes.
https://code.woboq.org/qt5/qtbase/src/corelib/kernel/qobject.cpp.html#_ZL14signalSlotLockPK7QObject

Reply:
But it can only cause audio skips if the engine is connecting and disconnecting signals frequently.

Reply:
My reasoning for not letting it cycle is that the number of connections to the guiTick20Ms CO will grow as the app is open longer, and I don't like having to solve bugs involving Mixxx being open for a while, so I'd prefer to not let the connection list to the VSync render callback grow indefinitely.

Reply:
How would the number of connections keep growing after the skin is initialized? I thought you were trying to avoid keeping a connection to every knob (which there are a lot of!) because a user with a mouse is only moving one at a time.

Reply:
> I am afraid that it can cause adible clicks, normally we do not reconnecting COs after initialisation.

Controller scripts that do layer switching do plenty of this.

Reply:
OK so it's fine. Thank you for refreshing our knowledge.